mozci/util/classify_batch_execution.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+import argparse
+import csv
+import datetime
+import itertools
+import json
+import os
+import time
+import uuid
+from multiprocessing import Pool
+
+from loguru import logger
+
+from mozci.push import MAX_DEPTH, Push, make_push_objects
+
+BASE_OUTPUT_DIR = (
+    os.path.dirname(os.path.realpath(__file__)) + "/classify_batch_results"
+)
+DAYS_FROM_TODAY = 30
+PARAMETERS_NAMES = [
+    "intermittent_confidence_threshold",
+    "real_confidence_threshold",
+    "use_possible_regressions",
+    "unknown_from_regressions",
+    "consider_children_pushes_configs",
+    "cross_config_counts",
+    "consistent_failures_counts",
+]
+PARAMETERS_COMBINATIONS = [
+    dict(zip(PARAMETERS_NAMES, parameters_values))
+    for parameters_values in itertools.product(
+        [0.5, 0.7, 0.9],  # intermittent_confidence_threshold
+        [0.7, 0.8, 0.9],  # real_confidence_threshold
+        [True, False],  # use_possible_regressions
+        [True, False],  # unknown_from_regressions
+        [True, False],  # consider_children_pushes_configs
+        [(2, 2), (2, 3), (2, 5)],  # cross_config_counts
+        [(2, 3), (2, 5)],  # consistent_failures_counts
+    )
+]
+CSV_HEADER = (
+    ["run_uuid", "push_uuid"]
+    + PARAMETERS_NAMES
+    + ["classification", "time_spent", "now"]
+)
+MAXIMUM_PROCESSES = os.cpu_count()
+
+
+def retrieve_pushes():
+    now = datetime.datetime.today()
+    to_date = now.strftime("%Y-%m-%d")
+    from_date = (now - datetime.timedelta(days=DAYS_FROM_TODAY)).strftime("%Y-%m-%d")
+    pushes = make_push_objects(from_date=from_date, to_date=to_date, branch="autoland")
+    return [{"rev": push.rev, "branch": push.branch} for push in pushes]
+
+
+def prepare_for_analysis(push):
+    push = Push(push["rev"], branch=push["branch"])
+
+    all_pushes = set(
+        [push]
+        + [parent for parent in push._iterate_parents(max_depth=MAX_DEPTH)]
+        + [child for child in push._iterate_children(max_depth=MAX_DEPTH)]
+    )
+    for p in all_pushes:
+        # Ignore retriggers and backfills on current push/its parents/its children.
+        p.tasks = [
+            task for task in p.tasks if not task.is_backfill and not task.is_retrigger
+        ]
+
+        # Pretend push was not backed out.
+        p.backedoutby = None
+
+        # Pretend push was not finalized yet.
+        p._date = datetime.datetime.now().timestamp()
+
+        # Pretend no tasks were classified to run the model without any outside help.
+        for task in p.tasks:
+            task.classification = "not classified"
+            task.classification_note = None
```

Comment:
This part too, if we make adjustments in the eval command we have to remember to make the same adjustments here too.

Reply:
Sorry, I meant to mention the other comment I made around code duplication.
It'd be nice to deduplicate the code between the script and eval, so we can share it and we don't have to remember to replicate the improvements we make to one to the other.